libnavigation-base/src/main/java/com/mapbox/navigation/base/utils/route/RouteProgressEx.kt

```diff
@@ -8,35 +8,61 @@ import com.mapbox.api.directions.v5.models.RouteOptions
 import com.mapbox.geojson.Point
 import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
 import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.trip.model.RouteProgress
 import com.mapbox.navigation.base.utils.DecodeUtils.stepGeometryToPoints
+import com.mapbox.navigation.utils.internal.ifNonNull
 import com.mapbox.navigation.utils.internal.logD
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
 
 private const val LOG_CATEGORY = "NavigationRouteUtils"
 
 /**
- * This function checks whether the [NavigationRoute] has unexpected closures, which could be a reason to re-route.
+ * This function checks whether the [NavigationRoute] has unexpected upcoming closures, which could be a reason to re-route.
  *
- * `true` is returned whenever there's any [RouteLeg.closures] outside of a direct region around a coordinate from [RouteOptions.coordinatesList]
+ * The algorithm does not take to account current closures, where the puck is on.
+ *
+ * `true` is returned whenever there's any upcoming [RouteLeg.closures] outside of a direct region around a coordinate from [RouteOptions.coordinatesList]
  * that was allowed for snapping to closures with [RouteOptions.snappingIncludeClosuresList] or [RouteOptions.snappingIncludeStaticClosuresList].
  * flags. Otherwise, `false` is returned.
  */
 @ExperimentalPreviewMapboxNavigationAPI
-suspend fun NavigationRoute.hasUnexpectedClosures(): Boolean =
+suspend fun RouteProgress.hasUnexpectedUpcomingClosures(): Boolean =
     withContext(Dispatchers.Default) {
-        val snappingResultList = directionsRoute.getSnappingResultList()
+        val snappingResultList = navigationRoute.directionsRoute.getSnappingResultList()
+
+        val routeProgressData = ifNonNull(
+            currentLegProgress,
```

Comment:
Question which is outside of the scope of this PR.
Hm, interesting why current leg progress is nullable. As I see [we always create it](https://github.com/mapbox/mapbox-navigation-android/blob/1685bec68acc00af583f53949e21cb04aed89f95/libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/NavigatorMapper.kt#L166), don't we?

Reply:
I suppose it's in the scope actually. I had the same doubts, found that [RouteLegProgress](https://github.com/mapbox/mapbox-navigation-android/blob/1685bec68acc00af583f53949e21cb04aed89f95/libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/NavigatorMapper.kt#L166) and [RouteStepProgress](https://github.com/mapbox/mapbox-navigation-android/blob/1685bec68acc00af583f53949e21cb04aed89f95/libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/NavigatorMapper.kt#L154) are not null, so we should be good.